FWCore/Framework/src/WorkerManager.cc

```diff
@@ -126,14 +126,17 @@ namespace edm {
     }
   }
 
-  void WorkerManager::setupOnDemandSystem(Principal& ep, EventSetupImpl const& es) {
+  void WorkerManager::setupResolvers(Principal& ep) {
     this->resetAll();
-    unscheduled_.setEventSetup(es);
     if (&ep != lastSetupEventPrincipal_) {
       UnscheduledConfigurator config(allWorkers_.begin(), allWorkers_.end(), &(unscheduled_.auxiliary()));
       ep.setupUnscheduled(config);
```

Comment:
This stuff is still part of the 'on demand' system so the split of function names isn't really accurate.

Reply:
You are partially correct because this is important for the 'on demand' system, but it is also important for other things and would be needed even of all producers were scheduled.

resetAll affects all types of resolvers, not just the unscheduled resolvers.

Four types of ProductResolvers have data members which are set by the call to setupUnscheduled. For PuttableProductResolvers, this sets worker_, which is used to notify other modules when a scheduled module is finished prefetching. This would be needed even if all producers were scheduled. For InputProductResolvers, it is setting up signals related to delayed read. I agree that in the case of UnscheduledProductResolvers it is part of the "on demand" system. I don't know how SwitchBaseProductResolver works, so I am not sure whether that is related to "on demand" or not.